repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java

```diff
@@ -446,16 +446,21 @@ private void traverseEdges(AtlasVertex datasetVertex, boolean isInput, int depth
 
             int qualifyingEdges = 0;
             for (AtlasEdge incomingEdge : processEdgesList) {
-                AtlasVertex         processVertex = incomingEdge.getOutVertex();
-                Iterator<AtlasEdge> outgoingEdges = processVertex.getEdges(OUT, isInput ? PROCESS_INPUTS_EDGE : PROCESS_OUTPUTS_EDGE).iterator();
-
-                while (outgoingEdges.hasNext()) {
-                    AtlasEdge edge = outgoingEdges.next();
-                    if (edge.getInVertex().equals(lineageContext.getStartDatasetVertex()) || lineageContext.evaluate(edge.getInVertex())) {
-                        qualifyingEdges++;
-                        break;
+                if (GraphHelper.getStatus(incomingEdge) == AtlasEntity.Status.ACTIVE) {
+
+                    AtlasVertex processVertex = incomingEdge.getOutVertex();
+                    Iterator<AtlasEdge> outgoingEdges = processVertex.getEdges(OUT, isInput ? PROCESS_INPUTS_EDGE : PROCESS_OUTPUTS_EDGE).iterator();
+
+                    while (outgoingEdges.hasNext()) {
+                        AtlasEdge edge = outgoingEdges.next();
+                        if (GraphHelper.getStatus(edge) == AtlasEntity.Status.ACTIVE && (edge.getInVertex().equals(lineageContext.getStartDatasetVertex())
```

Comment:
Let's use isAllowDeletedProcess flag instead of this, we already this flag at another place

Reply:
Added isAllowDeletedProcess flag check with ACTIVE edge check